galsim/utilities.py

```diff
@@ -571,20 +679,31 @@ def interleaveImages(im_list, N, offsets, add_flux=True, suppress_warnings=False
         img /= 1.0*len(im_list)
 
     # Assign an appropriate WCS for the output
-    if scale is not None:
-        if n1==n2:
-            img.wcs = galsim.PixelScale(1.*scale/n1)
-        else:
-            img.wcs = galsim.JacobianWCS(1.*scale/n1, 0., 0., 1.*scale/n2)
-    elif isinstance(wcs,galsim.JacobianWCS): # from say, a previously interleaved Image.
-        dudx, dudy, dvdx, dvdy = wcs.dudx, wcs.dudy, wcs.dvdx, wcs.dvdy
-        if (1.0*dudx/n1==1.0*dvdy/n2) and (dudy==0.0) and (dvdx==0.0):
-            img.wcs = galsim.PixelScale(1.*dudx/n1)
+    if wcs is not None: # and wcs.isLocal():
```

Comment:
You at least need to check `if wcs.isUniform()`, since if it is not, the call to `wcs.jacobian()` without any position argument will raise an exception.  Probably better to fall through to your warning about being unable to assign a WCS automatically instead.

Likewise in `deinterleaveImage` above.